rclpy/rclpy/executors.py

```diff
@@ -410,9 +416,6 @@ def _wait_for_ready_callbacks(self, timeout_sec=None, nodes=None):
                 clients_ready = _rclpy.rclpy_get_ready_entities('client', wait_set)
                 services_ready = _rclpy.rclpy_get_ready_entities('service', wait_set)
 
-            # Check sigint guard condition
-            if sigint_gc_handle in guards_ready:
-                raise KeyboardInterrupt
```

Comment:
This case is no longer appropriate because the `KeyboardInterrupt` will be raised (if at all) from within `_rclpy_wait()` by the signal handler.